### PR TITLE
Update deploy-prd-enhanced-worker.yml

### PR DESCRIPTION
**Pull request by branberry**
Add additional files that trigger a build for the enhanced webhook and worker. If these directories are not included, it could cause errors such as missing environment variables. For example, if we add a secure string for the worker in `utils/ssm.ts`, that will not trigger a deploy for the worker infrastructure, leading to errors from the missing environment variable.

### CODE REVIEW

**Review by schmalliso — APPROVED** [review]

File: .github/workflows/deploy-prd-enhanced-worker.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 on:
   push:
-    paths: ["src/**", "cdk-infra/lib/constructs/worker/**"]
+    paths: ["src/**", "cdk-infra/lib/constructs/worker/**", "Dockerfile.enhanced", "cdk-infra/utils"]
```
Comment (rayangler) [inline .github/workflows/deploy-prd-enhanced-worker.yml:3]: Just a heads up that pushes for paths aren't evaluated for pushes to tags, based on the [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore). 🤔 

> Path filters are not evaluated for pushes of tags.

**Review by rayangler — APPROVED** [review]
LGTM! But just wanted to flag some behavior

File: .github/workflows/deploy-prd-enhanced-worker.yml (hunk 1 above)
Comment (branberry) [reply .github/workflows/deploy-prd-enhanced-worker.yml:3]: Good catch, @rayangler! I will just remove the paths for the `prd` actions which shouldn't be an issue

**Review by rayangler — APPROVED** [review]

File: .github/workflows/deploy-prd-enhanced-worker.yml (hunk 1 above)
Comment (rayangler) [reply .github/workflows/deploy-prd-enhanced-worker.yml]: Yeah, it's interesting behavior, but makes sense. We can definitely apply this path filter to a pre-prod environment in the future

**Pull request merged**